### PR TITLE
Add changelog for 4.0.1

### PR DESCRIPTION
**Pull request by pwschuurman**
What type of PR is this?
/kind documentation

What this PR does / why we need it: Changelog for patch release 4.0.1

Which issue(s) this PR fixes:
Fixes #

Special notes for your reviewer:

Does this PR introduce a user-facing change?:

```
NONE
```

### CODE REVIEW

**Comment by xing-yang** [comment]
@pwschuurman Can you run release-notes tool to generate the changelog and submit the PR directly to release-4.0?

**Comment by pwschuurman** [comment]
@xing-yang I updated the release-notes tool to re-generate the changelog, and updated to release-4.0 branch.

File: CHANGELOG/CHANGELOG-4.0.md [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
+[Documentation](https://kubernetes-csi.github.io)
+
+# Changelog since v4.0.0
+
+## Changes by Kind
+
+### Other (Cleanup or Flake)
+
+- --http-endpoint will now correctly be used for the metrics server address when --metrics-address is not provided. (#497, @ggriffiths)
+- Changed the webhook image from distroless/base to distroless/static. (#550, @WanzenBug)
```
Comment (xing-yang) [inline CHANGELOG/CHANGELOG-4.0.md:12]: This needs some manual work for the PR and git id link.  See the following example:
https://github.com/kubernetes-csi/external-snapshotter/pull/635/files

**Comment by pwschuurman** [comment]
@xing-yang Updated. I re-ran `release-notes` using the `--markdown-links` flag. That fixed the PR and gid id links

**Pull request merged**